app/models/bot/alegre.rb

```diff
@@ -490,7 +496,12 @@ def self.request_api(method, path, params = {}, query_or_body = 'body', retries
       response_body = response.body
       Rails.logger.info("[Alegre Bot] Alegre response: #{response_body.inspect}")
       ActiveRecord::Base.connection.reconnect! if RequestStore.store[:pause_database_connection]
-      JSON.parse(response_body)
+      parsed_response = JSON.parse(response_body)
+      if parsed_response.dig("queue") == 'audio__Model' && parsed_response.dig("body", "callback_url") != nil
+        redis = Redis.new(REDIS_CONFIG)
+        redis_response = redis.blpop(parsed_response.dig("body", "id"), timeout=120)
+        return JSON.parse(redis_response[1])
+      end
```

Comment:
Do we need a line `parsed_response` after `end` to ensure `parsed_response` is the return value if non-callback functions? (I always get confused with this part of Ruby)

Reply:
Let's please add a prefix to the Redis keys, just for better scoping/debugging if needed, like a namespace... so, instead of just `parsed_response.dig("body", "id")`, something like "alegre:webhook:#{parsed_response.dig("body", "id")}", for example.

Reply:
We do, @computermacgyver !

Reply:
Duh..... ok I will address in morning!